File: src/frontend/Portaal.Angular/src/app/Models/assignment.ts
import { Ratings } from "./ratings";
import { Tag } from "./tag";
import { Comment } from "./comment";

export class Assignment {
    id?: number;
    title: string = "title";
    description: string = "description";
    dateCreated?: Date;
    createById: number = -1;
    // attachments: Attachment[] = [];
    tags: Tag[] = [];
    ratings: Ratings = new Ratings;
    comments: Comment[] = [];
}

File: src/backend/PortaalBackend/PortaalBackend.Domain/Models/Assignment.cs
using PortaalBackend.Domain.Interfaces;
using PortaalBackend.Domain.Models.Joins;

namespace PortaalBackend.Domain.Models
{
    public record Assignment : IEntity
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime DateCreated { get; set; }
        public int CreatedById { get; set; }
        //public List<Attachment> Attachments { get; set; }
        public ICollection<AssignmentTag> AssignmentTags { get; set; } = new List<AssignmentTag>();
        public Ratings Ratings { get; set; } = new();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}
